File: Rust/Average-Salary-Excluding-Minimum-And-Maximum-Salary.rs
pub struct Solution {}

impl Solution {
    pub fn average(salary: Vec<i32>) -> f64 {
        let (mut min, mut max, mut sum) = (i32::MAX, 0, 0);

        for sal in salary.iter() {
            sum += sal;
            min = min.min(*sal);
            max = max.max(*sal);
        }

        (sum - min - max) as f64 / (salary.len() as f64 - 2.)
    }
}

fn main() {}

File: Go/Average-Salary-Excluding-Minimum-And-Maximum-Salary.go
package main

import "math"



func average(salary []int) float64 {
	var min float64 = float64(salary[0])
	var max float64 = 0;
	var sum float64 = 0

	for _, sal := range salary {
		sum += float64(sal)
		max = math.Max(float64(max), float64(sal));
		min = math.Min(min, float64(sal))
	}

	return (sum - min-max) / float64(len(salary) - 2)
}